### DISCUSSION TITLE
Is there a way to render a single document within a project?

### DISCUSSION THREAD

**Discussion by roaldarbol** [quarto-dev/quarto-cli#11091] [Q&A]
### Description

I've got a book project which has many parts. I'd like to send a single part/document to a colleague, rendered in `docx`. Even if I `quarto render my-single-file.qmd` it only renders the entire project according to `_quarto.yml`.

The file has its own rendering YAML settings:
```qmd
---
title: My title
abstract: Something should go here.
author:
  - name: Mikkel Roald-Arbøl
    affiliations:
      - name: University of Sussex
        department: School of Life Sciences
format: 
  docx:
    output-file: the-article.docx
---
```
Is this in any way possible in Quarto? (I'm used to this kind of workflow from MyST so just checking)

**Comment by mcanouil** [comment]
That's the designed/expected behaviour.
If a document is inside a project, then Quarto respects it, especially, if you render from the root of your project.

Make a temporary copy or move inside the directory where the document is.

**Reply by mcanouil** [reply]
Another thought, you could setup a profile which uses the "default" project type and activate that profile, see <https://quarto.org/docs/projects/profiles.html>.

**Reply by roaldarbol** [reply]
Yeah, I did imagine it to be expected behaviour too. 😊 I think Profiles might be a good suggestion for this use-case! I'll play around with it and report back. Thanks @mcanouil!

PS. Even if I `cd` into the correct directory, I still get the entire project rendered; presumably Quarto looks in lower folders for a `_quarto.yml` file before starting the rendering process.

**Reply by roaldarbol** [reply]
Copying to a different location works too. Only thing one needs to remember is that other files, e.g. bibliography that normally lives elsewhere in the project needs to be copied over too.

**Reply by mcanouil** [reply]
> PS. Even if I cd into the correct directory, I still get the entire project rendered; presumably Quarto looks in lower folders for a _quarto.yml file before starting the rendering process.

Yes, I was expected this but worth checking as I don't know well (or at all) the "project detection" algorithm.

**Reply by mcanouil** [reply]
> Copying to a different location works too. Only thing one needs to remember is that other files, e.g. bibliography that normally lives elsewhere in the project needs to be copied over too.

Indeed, it's why after I commented I thought of profiles which seem better in this case as you don't need to worry about file paths and resources.

**Reply by roaldarbol** [reply]
> Yes, I was expected this but worth checking as I don't know well (or at all) the "project detection" algorithm.

Same. Now we know. 😉
